src/main/java/io/vertx/core/impl/VertxImpl.java

```diff
@@ -406,6 +409,15 @@ public DnsClient createDnsClient(int port, String host) {
     return new DnsClientImpl(this, port, host, DnsClientOptions.DEFAULT_QUERY_TIMEOUT);
   }
 
+  @Override
+  public DnsClient createDnsClient() {
```

Comment:
actually it should use `DnsResolverProvider` so it can be configured using `VertxOptions#addressResolverOptions` (that's how we can test it).

Reply:
In `DnsResolverProvider`s constructor, we have a `serverList` so the list of DNS servers to use, but when creating DnsClient we are specifying only one sever. Should we create `DnsResolverProvider` with `addressResolverOptions` and then use the first element of serverList as a default DNS server with port=53?

Reply:
I have just updated the PR. Is that what you mean?

Reply:
yes, we should add a method on DnsResolverProvider that provides the server address (the first one) and have only the stream nameServerAddresses to be a field of DnsResolverProvider with a method that returns it, so when building the client, it can use the first one rather than using a List